bessctl/commands.py

```diff
@@ -194,6 +194,19 @@ def get_var_attrs(cli, var_token, partial_word):
             var_desc = 'configuration filename'
             var_candidates = complete_filename(partial_word)
 
+        if var_token == '[DIRECTION]':
+            var_type = 'name'
+            var_desc = 'gate direction discriminator (default "out")'
+            var_candidates = ['in', 'out']
+
+        if var_token == '[PRIORITY]':
```

Comment:
It might be conceptually simpler each hook type has its own pre-defined priority value, rather than user specifies it for each hook individually. One possible (common?) way to do this is:

- Suppose there are three hook types: x, y, z (x has the highest priority, z is the lowest)
- Module M has those hooks enabled for both its input gate and output gate
- Then the order of execution should be: x - y - z - M - z - y - x
  - It is something like destructors are called in the reverse order of constructors for a derived class in C++.